homeassistant/components/mqtt/device_trigger.py

```diff
@@ -193,16 +194,16 @@ def __init__(
         hass: HomeAssistant,
         config: ConfigType,
         device_id: str,
-        discovery_data: dict,
+        discovery_data: DiscoveryInfoType,
         config_entry: ConfigEntry,
     ) -> None:
         """Initialize."""
-        self._config = config
-        self._config_entry = config_entry
-        self.device_id = device_id
-        self.discovery_data = discovery_data
-        self.hass = hass
-        self._mqtt_data = get_mqtt_data(hass)
+        self._config: ConfigType = config
+        self._config_entry: ConfigEntry = config_entry
+        self.device_id: str = device_id
+        self.discovery_data: DiscoveryInfoType = discovery_data
+        self.hass: HomeAssistant = hass
+        self._mqtt_data: MqttData = get_mqtt_data(hass)
```

Comment:
These are inferred from the argument type, so there is no need to set them again.
```suggestion
        self._config = config
        self._config_entry = config_entry
        self.device_id = device_id
        self.discovery_data = discovery_data
        self.hass = hass
        self._mqtt_data = get_mqtt_data(hass)
```